src/main/java/digital/slovensko/autogram/core/TargetPath.java

```diff
@@ -0,0 +1,141 @@
+package digital.slovensko.autogram.core;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+import com.google.common.io.Files;
+
+import digital.slovensko.autogram.core.errors.AutogramException;
+
+public class TargetPath {
+    private final File targetDirectory;
+    private final String targetName;
+    private final File sourceFile;
+    private final boolean isForce;
+    private final boolean isGenerated;
+    private final boolean isForMultipleFiles;
+
+    public TargetPath(String target, File source, boolean force) {
+        this.sourceFile = source;
+        this.isForce = force;
+
+        // TODO toto nejako otestovat
+        this.isGenerated = target == null;
+        this.isForMultipleFiles = source.isDirectory();
+        if (isGenerated) {
+            this.targetDirectory = isForMultipleFiles ? new File(source.getPath() + "_signed")
+                    : source.getParentFile();
+            this.targetName = null;
+        } else {
+            var targetFile = new File(target);
+            if (!hasSourceAndTargetMatchingType(sourceFile, targetFile)) {
+                throw new IllegalArgumentException("Source and target incompatible file types");
+            }
+            this.targetDirectory = source.isDirectory() ? targetFile : targetFile.getParentFile();
+            this.targetName = targetFile.getName();
+        }
+    }
+
+    public static TargetPath fromParams(CliParameters params) {
+        return new TargetPath(params.getTarget(), params.getSource(), params.isForce());
+    }
+
+    public static TargetPath fromSource(File source) {
+        return new TargetPath(null, source, false);
+    }
+
+    private TargetPath(File source, boolean force, boolean isGenerated, File targetDirectory, String targetName) {
+        this.sourceFile = source;
+        this.isForce = force;
+        this.isGenerated = isGenerated;
+        this.isForMultipleFiles = source.isDirectory();
+        this.targetDirectory = targetDirectory;
+        this.targetName = targetName;
+    }
+
+    /**
+     * use this constructor only for testing
+     */
+    public static TargetPath buildForTest(File source, boolean force, boolean isGenerated, File targetDirectory,
+            String targetName) {
+        return new TargetPath(source, force, isGenerated, targetDirectory, targetName);
+    }
```

Comment:
@pomali môžem dať toto priamo do testu a ten konštruktor dáme public?

Reply:
Naschvál som to dal takto, aby bolo jasné, že to nie je na bežné použitie.

Reply:
ešte pozerám, či vôbec potrebujeme a chceme špeciálny konštruktor pre testy. Nemôžeme použiť normálne tie fromParams a fromSource? Tým pádom sa pretestuje aj tá ďalšia logika rozhodovania.

Reply:
Je to aby som podhodil mock File, ak máš lepší nápad ako to urobiť tak som za